### PR TITLE
Add obb intersection check

### PR DESCRIPTION
**Pull request by eisoku9618**
add an OBB intersection check function using `Separating Axis Theorem` proposed in the following paper.

```cite
S. Gottschalk, M.C. Lin, D. Manocha. OBBTree: A Hierarchical Structure for Rapid Interference Detection. In Proceedings of SIGGRAPH '96.
```

- unit test: `testopenrave:add-obb-intersection-check`
- Pipeline #673913

### CODE REVIEW

**Review by yoshikikanemoto — APPROVED** [review]
LGTM. @rdiankov can you check and merge?

File: include/openrave/geometry.h [hunk 1]
```diff
@@ -2002,125 +1996,111 @@ inline bool AABBCollision(const RaveAxisAlignedBox<T>& ab1, const RaveAxisAligne
...
+{
+    const RaveTransform<T> relativeTransform(obb.transform.rot, obb.transform.trans - ab.pos);
+    return BoxAtOriginOBBCollision(ab.extents, RaveOrientedBox<T>(relativeTransform, obb.extents));
+}
+
+/// \brief Test collision between an axis-aligned bounding box located at world origin and an oriented bounding box.
+/// using Separating Axis Theorem.
+/// S. Gottschalk, M.C. Lin, D. Manocha. OBBTree: A Hierarchical Structure for Rapid Interference Detection. In Proceedings of SIGGRAPH '96.
+template <typename T>
+inline bool BoxAtOriginOBBCollision(const RaveVector<T>& extents, const RaveOrientedBox<T>& obb)
```
Comment (rdiankov) [inline include/openrave/geometry.h:2019]: Given these are new functions, perhaps you can give them better names, ones that start with a verb like Check?

Comment (eisoku9618) [reply include/openrave/geometry.h]: @rdiankov Thank you, improved!

File: python/bindings/openravepy_global.cpp [hunk 2]
```diff
@@ -1331,6 +1331,36 @@ object TransformLookat(object olookat, object ocamerapos, object ocameraup)
...
+{
+    return toPyOrientedBox(OrientedBoxFromAABB(ExtractAABB(oab), ExtractTransform(otransform)));
+}
+
+object AABBFromOrientedBox(object oobb)
+{
+    return toPyAABB(AABBFromOrientedBox(ExtractOrientedBox(oobb)));
+}
+
+bool AABBCollision(object oab1, object oab2)
```
Comment (yoshikikanemoto) [inline python/bindings/openravepy_global.cpp:1344]: @eisoku9618 let's also change name of this function. and also do the same to c++. and mark the existing one obsolete

Comment (eisoku9618) [reply python/bindings/openravepy_global.cpp:1350]: @yoshikikanemoto sure, I added `CheckAABBCollision()` and appended deprecated attribute to existing `AABBCollision()`.

Comment (yoshikikanemoto) [reply python/bindings/openravepy_global.cpp:1350]: can we remove `AABBCollision` from python bindings since it's not there before this PR?

Comment (eisoku9618) [reply python/bindings/openravepy_global.cpp]: Yes, thank you!

File: include/openrave/geometry.h (hunk 1 above)
Comment (yoshikikanemoto) [reply include/openrave/geometry.h]: @rdiankov i think it's good again now.

**Comment by rdiankov** [comment]
thanks

**Pull request merged**